src/Task/SymfonyConsole.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Formatter\ProcessFormatterInterface;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Config\ConfigOptionsResolver;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/** @extends AbstractExternalTask<ProcessFormatterInterface> */
+class SymfonyConsole extends AbstractExternalTask
+{
+    public static function getConfigurableOptions(): ConfigOptionsResolver
+    {
+        return ConfigOptionsResolver::fromOptionsResolver(
+            (new OptionsResolver())
+                ->setDefaults([
+                    'bin' => './bin/console',
+                    'command' => [],
+                ])
+                ->addAllowedTypes('command', ['string[]'])
+                ->setRequired('command')
+        );
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
+        if (0 === \count($context->getFiles())) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        if ('' === \implode('', $config['command'])) {
+            return TaskResult::createNonBlockingFailed(
```

Comment:
maybe it makes more sense to validate this during option resolving?

```
$resolver->setAllowedValues('command', function ($value): bool {
    // return true or false
});
```

Reply:
Implemented, 
the only drawback i.m.h.o. with this approach is that the message is a bit less clear and the command fails blocking on config.